File: 2022_c#_basics/12-while-loop-exercise/01-old-books.cs
using System;

namespace _01_Old_Books
{
    internal class Program
    {
        static void Main(string[] args)
        {
            string searchedBook = Console.ReadLine();
            string book = Console.ReadLine();
            int checkedBooks = 0;
            int foundBook = 0;

            while (book != "No More Books")
            {
                if (book == searchedBook)
                {
                    foundBook += 1;

                    if (foundBook > 0)
                    {
                        Console.WriteLine($"You checked {checkedBooks} books and found it.");
                        break;
                    }

                }
                checkedBooks += 1;
                book = Console.ReadLine();
            }

            if (foundBook == 0)
            {
                Console.WriteLine($"The book you search is not here!");
                Console.WriteLine($"You checked {checkedBooks} books.");
            }
        }
    }
}

File: 2024_java_basics/12-while-loops-lab/tasks/01-old-books.java
import java.util.Scanner;

public class OldBooks {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String searchedBook = scanner.nextLine();
        String book = scanner.nextLine();
        
        int checkedBooks = 0;
        boolean foundBook = false;

        while (!book.equals("No More Books")) {
            if (book.equals(searchedBook)) {
                foundBook = true;
                System.out.println("You checked " + checkedBooks + " books and found it.");
                break;
            }
            checkedBooks++;
            book = scanner.nextLine();
        }

        if (!foundBook) {
            System.out.println("The book you search is not here!");
            System.out.println("You checked " + checkedBooks + " books.");
        }

        scanner.close();
    }
}
